nautobot/cloud/filters.py

```diff
@@ -146,4 +150,27 @@ class CloudServiceFilterSet(NautobotFilterSet):
 
     class Meta:
         model = models.CloudService
-        fields = ["id", "name", "description", "cloud_account", "cloud_network", "cloud_resource_type", "tags"]
+        fields = ["id", "name", "description", "cloud_account", "cloud_networks", "cloud_resource_type", "tags"]
+
+
+class CloudServiceNetworkAssignmentFilterSet(BaseFilterSet):
+    q = SearchFilter(
+        filter_predicates={
+            "cloud_network__name": "icontains",
+            "cloud_network__description": "icontains",
+            "cloud_service__name": "icontains",
+            "cloud_service__description": "icontains",
+        }
```

Comment:
Just wondering - should this include all of the nested lookups from `CloudServiceFilterSet.q` and `CloudNetworkFilterSet.q`? It feels slightly wrong to me that a search term that matches a CloudService will not necessarily match each its CloudServiceNetworkAssignments, but I don't feel strongly.